examples/helpers/glium.rs

```diff
@@ -0,0 +1,121 @@
+
+pub struct GliumDrawer<'a, F: 'a> {
+    display: &'a F,
+    vertex_buffer: glium::VertexBuffer<Vertex>,
+    index_buffer: glium::IndexBuffer<u16>,
+    program: glium::Program,
+}
+
+impl<'a, F: glium::backend::Facade + 'a> GliumDrawer<'a, F> {
+    pub fn new(display: &'a F) -> GliumDrawer<'a, F> {
+
+        // building the vertex buffer, which contains all the vertices that we will draw
+        let vertex_buffer = glium::VertexBuffer::new(display,
+                                                     &[Vertex {
+                                                           position: [0.0, 0.0],
+                                                           tex_coords: [0.0, 0.0],
+                                                       },
+                                                       Vertex {
+                                                           position: [0.0, 1.0],
+                                                           tex_coords: [0.0, 1.0],
+                                                       },
+                                                       Vertex {
+                                                           position: [1.0, 1.0],
+                                                           tex_coords: [1.0, 1.0],
+                                                       },
+                                                       Vertex {
+                                                           position: [1.0, 0.0],
+                                                           tex_coords: [1.0, 0.0],
+                                                       }]).unwrap();
+
+        // building the index buffer
+        let index_buffer =
+            glium::IndexBuffer::new(display, PrimitiveType::TriangleStrip, &[1 as u16, 2, 0, 3]).unwrap();
+
+        // compiling shaders and linking them together
+        let program = program!(display,
+			100 => {
+				vertex: "
+					#version 100
+					uniform lowp mat4 matrix;
+					attribute lowp vec2 position;
+					attribute lowp vec2 tex_coords;
+					varying lowp vec2 v_tex_coords;
+					void main() {
+						gl_Position = matrix * vec4(position, 0.0, 1.0);
+						v_tex_coords = tex_coords;
+					}
+				",
+
+				fragment: "
+					#version 100
+					uniform lowp sampler2D tex;
+					varying lowp vec2 v_tex_coords;
+					void main() {
+	                    lowp vec4 color = texture2D(tex, v_tex_coords);
+						gl_FragColor.r = color.z;
+                        gl_FragColor.g = color.y;
+                        gl_FragColor.b = color.x;
+                        gl_FragColor.a = color.w;
+					}
+				",
+			},
+        ).unwrap();
+
+        GliumDrawer {
+            display,
+            vertex_buffer,
+            index_buffer,
+            program,
+        }
+    }
+
+    pub fn draw(&self, target: &mut glium::Frame, contents: &[u8], surface_dimensions: (u32, u32),
+                surface_location: (i32, i32), screen_size: (u32, u32)) {
```

Comment:
While this will likely will never be a performance bottleneck in our code, given the amount of surfaces we will have at once on the screen, you would usually try to minimize the amount of draw calls and instead commit all vertices and textures at once.
So we would instead cache all contents to be drawn and then have a "finalize" call or something similar, that results in once draw call with all windows. However this would make this code unnecessarily complex, I think. Might be worth to benchmark this some time later.